dist/tools/format-board-list/format-board-list.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+import os
+import sys
+import subprocess
+import re
+from pprint import pprint
+from tempfile import NamedTemporaryFile
+import shutil
+
+TARGET_VARIABLES = ['BOARD_INSUFFICIENT_MEMORY', 'BOARD_BLACKLIST', 'BOARD_WHITELIST']
+
+def chdir_to_riot_root():
+	SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+	RIOT_DIR = os.path.realpath(SCRIPT_DIR + "/../../..")
+	os.chdir(RIOT_DIR)
+
+
+class BoardList:
+	def __init__(self, first_line_of_board_list):
+		self.boards = []
+		self.target_variable = None
+		self.variable_operation = None
+		self.regex = re.compile('([A-Z_]+)\\s*([\\+:]?=)(.+)')
+		self.parse_first_line(first_line_of_board_list)
+
+	def parse_first_line(self, line):
+		m = self.regex.match(line)
+		assert m is not None, "Regex should match '%s'" % (line)
+		assert m.group(1) in TARGET_VARIABLES, "Match '%s' should be in TARGET_VARIABLES" % (m.group(1))
+
+		self.target_variable = m.group(1)
+		self.variable_operation = m.group(2)
+		self.add_boards_from_line(m.group(3))
+
+	def add_boards_from_line(self, line):
+		boards = self._extract_boards(line)
+		self.boards.extend(boards)
+
+	def _extract_boards(self, line):
+		cleaned_line = line.strip().rstrip('\\').rstrip()
+		boards = cleaned_line.split(' ')
+		boards = list(map(lambda board: board.strip(), boards))
+		boards = list(filter(lambda board: len(board) > 0, boards))
+		return boards
+
+	def __str__(self):
+		self.boards.sort()
+		prefix = "%s %s " % (self.target_variable, self.variable_operation)
+		spacing = len(prefix) * ' '
+		return prefix + (" \\\n" + spacing).join(self.boards)
+
+
+def get_affected_filenames(target_variable):
```

Comment:
Is it acceptable to have both the formatting logic and the file selection logic here?

Reply:
I don't think I understand the comment, could you elaborate?

Reply:
What happens if I want to auto-format a single Makefile? For example, I want to fix a file before committing it.

Reply:
Ah I see. I've been simply running it over the entire code base. It's not too slow, so it hasn't been bothering me. Also has the added benefit of me not being able to forget any Makefile I might've touched 😅

Reply:
There are several static checks with the same issue, so it's not a huuuge problem if this script does the same, though at some point it would be nice to have them fixed, but that's an issue for another PR.